Fix code quality issues

## Summary

The code base requires refactoring to address multiple style and typehinting issues identified by static analysis tools. These issues include:

- **PHPCS Findings (from Report 1)**:
  - Exceeding line-length standards in `/src/README.md`.
  - Concatenation of translatable strings, non-literal strings passed to `t()`, and backslash escaping in translatable strings in `/src/src/Form/AiAgentFormTrait.php`.

- **Drupal-check/PHPStan Findings (from Report 2)**:
  - Methods and parameters lack explicit typehints (especially array value types).
  - Some methods have no declared return type.
  - Incorrect usage of `TranslatableMarkup` with functions expecting plain strings.
  - Potential logic issues indicated by PHPStan (e.g., "If condition is always true").

Refactoring these parts of the code will help ensure the codebase adheres to Drupal coding standards, improves overall maintainability, and reduces the likelihood of runtime errors.

**Full Reports:**

**Report 1 (PHPCS):**
```
FILE: /src/README.md
----------------------------------------------------------------------
FOUND 0 ERRORS AND 9 WARNINGS AFFECTING 9 LINES
----------------------------------------------------------------------
   5 | WARNING | Line exceeds 80 characters; contains 225 characters
  17 | WARNING | Line exceeds 80 characters; contains 82 characters
  24 | WARNING | Line exceeds 80 characters; contains 125 characters
  26 | WARNING | Line exceeds 80 characters; contains 107 characters
  56 | WARNING | Line exceeds 80 characters; contains 167 characters
  61 | WARNING | Line exceeds 80 characters; contains 89 characters
  67 | WARNING | Line exceeds 80 characters; contains 89 characters
  94 | WARNING | Line exceeds 80 characters; contains 88 characters
 195 | WARNING | Line exceeds 80 characters; contains 86 characters
----------------------------------------------------------------------

FILE: /src/src/Form/AiAgentFormTrait.php
-------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 3 WARNINGS AFFECTING 4 LINES
-------------------------------------------------------------------------------------------------------------------------
  74 | ERROR   | Concatenating translatable strings is not allowed, use placeholders instead and only one string literal
 113 | WARNING | Only string literals should be passed to t() where possible
 114 | WARNING | Avoid backslash escaping in translatable strings when possible, use "" quotes instead
 201 | WARNING | Only string literals should be passed to t() where possible
-------------------------------------------------------------------------------------------------------------------------

Time: 901ms; Memory: 10MB
```

**Report 2 (Drupal-check/PHPStan):**
```
Using version ^1.5 for mglaman/drupal-check

# Run drupal-check with stderr redirected
./vendor/bin/drupal-check --drupal-root . -ad web/modules/contrib/ckeditor_ai_agent 2>/dev/null
 ------ -------------------------------------------------------------------------------------- 
  Line   /src/ckeditor_ai_agent.module                                                         
 ------ -------------------------------------------------------------------------------------- 
  13     Function ckeditor_ai_agent_ckeditor5_plugin_info_alter() has                          
         parameter $plugin_definitions with no value type specified in                         
         iterable type array.                                                                  
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  27     If condition is always true.                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this                    
            check by setting treatPhpDocTypesAsCertain: false in your                          
            /tmp/drupal_check_phpstan_1734699561.neon.                                         
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/AiAgentConfigurationManager.php                                              
 ------ -------------------------------------------------------------------------------------- 
  39     Method                                                                                
         Drupal\ckeditor_ai_agent\AiAgentConfigurationManager::getCkEditorConfig()             
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/AiAgentFormTrait.php (in context of class                               
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm)                                    
 ------ -------------------------------------------------------------------------------------- 
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getCommonFormElements()            
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::addPromptSettings()                
         has no return type specified.                                                         
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::addPromptSettings()                
         has parameter $config with no type specified.                                         
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::addPromptSettings()                
         has parameter $elements with no type specified.                                       
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::addPromptSettings()                
         has parameter $getConfigValue with no type specified.                                 
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::addPromptSettings()                
         has parameter $is_plugin with no type specified.                                      
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/AiAgentFormTrait.php (in context of class                               
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent)                              
 ------ -------------------------------------------------------------------------------------- 
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getCommonFormElements()      
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::addPromptSettings()          
         has no return type specified.                                                         
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::addPromptSettings()          
         has parameter $config with no type specified.                                         
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::addPromptSettings()          
         has parameter $elements with no type specified.                                       
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::addPromptSettings()          
         has parameter $getConfigValue with no type specified.                                 
  291    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::addPromptSettings()          
         has parameter $is_plugin with no type specified.                                      
  322    Parameter #1 $string of function strtolower expects string,                           
         Drupal\Core\StringTranslation\TranslatableMarkup given.                               
  330    Parameter #1 $string of function strtolower expects string,                           
         Drupal\Core\StringTranslation\TranslatableMarkup given.                               
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/AiAgentSettingsForm.php                                                 
 ------ -------------------------------------------------------------------------------------- 
  38     Method Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::create()                    
         has no return type specified.                                                         
  54     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getEditableConfigNames()           
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  61     Method Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::buildForm()                 
         has parameter $form with no value type specified in iterable type                     
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  61     Method Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::buildForm()                 
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  95     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::validateForm() has                 
         no return type specified.                                                             
  95     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::validateForm() has                 
         parameter $form with no value type specified in iterable type array.                  
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  108    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::submitForm() has                   
         no return type specified.                                                             
  108    Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::submitForm() has                   
         parameter $form with no value type specified in iterable type array.                  
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/ConfigMappingTrait.php (in context of class                             
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm)                                    
 ------ -------------------------------------------------------------------------------------- 
  16     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getConfigMapping()                 
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  99     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getPromptComponents()              
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/ConfigMappingTrait.php (in context of class                             
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent)                              
 ------ -------------------------------------------------------------------------------------- 
  16     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getConfigMapping()           
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  99     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getPromptComponents()        
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/ConfigSetterTrait.php (in context of class                              
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm)                                    
 ------ -------------------------------------------------------------------------------------- 
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processConfigValues()              
         has parameter $mapping with no value type specified in iterable type                  
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processConfigValues()              
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processConfigValues()              
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  39     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getNestedValue()                   
         has no return type specified.                                                         
  39     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::getNestedValue()                   
         has parameter $array with no value type specified in iterable type                    
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  56     Method Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::castValue()                 
         has no return type specified.                                                         
  56     Method Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::castValue()                 
         has parameter $value with no type specified.                                          
  82     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processModerationSettings()        
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  82     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processModerationSettings()        
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  98     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processPromptSettings()            
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  98     Method                                                                                
         Drupal\ckeditor_ai_agent\Form\AiAgentSettingsForm::processPromptSettings()            
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Form/ConfigSetterTrait.php (in context of class                              
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent)                              
 ------ -------------------------------------------------------------------------------------- 
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processConfigValues()        
         has parameter $mapping with no value type specified in iterable type                  
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processConfigValues()        
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  21     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processConfigValues()        
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  39     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getNestedValue()             
         has no return type specified.                                                         
  39     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getNestedValue()             
         has parameter $array with no value type specified in iterable type                    
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  56     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::castValue()                  
         has no return type specified.                                                         
  56     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::castValue()                  
         has parameter $value with no type specified.                                          
  82     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processModerationSettings()  
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  82     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processModerationSettings()  
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  98     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processPromptSettings()      
         has parameter $values with no value type specified in iterable type                   
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  98     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::processPromptSettings()      
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   /src/src/Plugin/CKEditor5Plugin/AiAgent.php                                           
 ------ -------------------------------------------------------------------------------------- 
  31     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::defaultConfiguration()       
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  78     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::buildConfigurationForm()     
         has parameter $form with no value type specified in iterable type                     
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  78     Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::buildConfigurationForm()     
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  129    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::submitConfigurationForm()    
         has no return type specified.                                                         
  129    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::submitConfigurationForm()    
         has parameter $form with no value type specified in iterable type                     
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  145    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getDynamicPluginConfig()     
         has parameter $static_plugin_config with no value type specified in                   
         iterable type array.                                                                  
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  145    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::getDynamicPluginConfig()     
         return type has no value type specified in iterable type array.                       
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
  216    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::validateConfigurationForm()  
         has no return type specified.                                                         
  216    Method                                                                                
         Drupal\ckeditor_ai_agent\Plugin\CKEditor5Plugin\AiAgent::validateConfigurationForm()  
         has parameter $form with no value type specified in iterable type                     
         array.                                                                                
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 

 [ERROR] Found 60 errors
```

## Improvements

- **Alignment with Drupal Coding Standards:** Addressing long lines, translatable string concatenation, and `t()` usage ensures compliance with recommended coding practices.
- **Improved Type Safety and Readability:** Adding explicit return types, parameter types, and iterable value types will make the codebase more self-documenting, maintainable, and less prone to subtle bugs.
- **Enhanced Maintenance and Future Proofing:** By resolving these issues now, future developers can more easily understand and extend the code, and static analysis tools can prevent similar issues going forward.

## Risks

- **Potential API Changes:** Adding strict typehints and return types may break backward compatibility if other code relies on the old signatures.
- **Refactoring Overhead:** The changes may require adjustments in multiple files and could introduce temporary instability if not carefully tested.
- **String Handling Adjustments:** Modifying how translation strings are managed may require review to ensure UI elements are still properly displayed.

## Involved components

Primarily, the following files (mentioned in the reports) will be updated:

- `/src/README.md` (line-length violations)
- `/src/src/Form/AiAgentFormTrait.php`
- `/src/ckeditor_ai_agent.module`
- `/src/src/AiAgentConfigurationManager.php`
- `/src/src/Form/AiAgentSettingsForm.php`
- `/src/src/Form/ConfigMappingTrait.php`
- `/src/src/Form/ConfigSetterTrait.php`
- `/src/src/Plugin/CKEditor5Plugin/AiAgent.php`

Basically, most or all files under `src/` as mentioned in the reports.

## Optional: Intended side effects

- **Better Developer Onboarding:** Clearer code and strict typehinting reduce learning curves for new contributors.
- **Increased Tooling Support:** Improved static analysis results and integration with IDE tools due to clearer type information.

## Optional: Missing test coverage

- **Tests for Refactored Methods:** Additional or adjusted unit tests may be needed to ensure that all methods with newly defined types behave as expected.
- **Functional Tests for Translations and UI:** Confirm that changes to translatable strings and UI logic do not negatively affect end-user experience.

--- a/README.md
+++ b/README.md
@@ -2,7 +2,9 @@
 
 ## Overview
 
-CKEditor AI Agent is a Drupal module that integrates AI-powered content generation capabilities into CKEditor 5. It provides an intuitive interface for generating, modifying, and enhancing content directly within your editor.
+CKEditor AI Agent is a Drupal module that integrates AI-powered content
+generation capabilities into CKEditor 5. It provides an intuitive interface for
+generating, modifying, and enhancing content directly within your editor.
 
 ## Requirements
 
@@ -14,16 +16,19 @@ CKEditor AI Agent is a Drupal module that integrates AI-powered content generati
 ## Installation
 
 1. **Install the Module**
-   - Download and place the module in your Drupal installation's modules directory
+   - Download and place the module in your Drupal installation's modules
+   directory
    - Enable the module through Drupal's admin interface or using Drush:
      ```bash
      drush en ckeditor_ai_agent
      ```
 
 2. **Configure CKEditor Integration**
-   - Go to **Administration > Configuration > Content authoring > Text formats and editors** (`admin/config/content/formats`)
+   - Go to **Administration > Configuration > Content authoring > Text formats
+   and editors** (`admin/config/content/formats`)
    - Edit your desired text format (typically Full HTML)
-   - Drag and drop the "AI Agent" button into the CKEditor toolbar to make it available for content editors
+   - Drag and drop the "AI Agent" button into the CKEditor toolbar to make it
+   available for content editors
 
 3. **Development**
 
@@ -51,7 +56,9 @@ CKEditor AI Agent is a Drupal module that integrates AI-powered content generati
 
 ## Configuration
 
-Navigate to `Administration > Configuration > Content authoring > CKEditor AI Agent Settings` (`/admin/config/content/ckeditor-ai-agent`) to configure global settings.
+Navigate to `Administration > Configuration > Content authoring > CKEditor AI
+Agent Settings` (`/admin/config/content/ckeditor-ai-agent`) to configure global
+settings.
 
 | Option | Type | Default | Description |
 |--------|------|---------|-------------|
@@ -144,4 +151,5 @@ For bug reports and feature requests, please use the [issue queue](https://www.d
 
 ## License
 
-This project is licensed under the GPL-2.0+ license. See the LICENSE file for details.
+This project is licensed under the GPL-2.0+ license. See the LICENSE file for
+details.

--- a/ckeditor_ai_agent.module
+++ b/ckeditor_ai_agent.module
@@ -6,6 +6,7 @@
  */
 
 use Drupal\ckeditor5\Plugin\CKEditor5PluginDefinition;
+use Drupal\Core\Routing\RouteMatchInterface;
 
 /**
  * Implements hook_ckeditor5_plugin_info_alter().
@@ -26,7 +27,8 @@ function ckeditor_ai_agent_ckeditor5_plugin_info_alter(array &$plugin_definition
 
     // Get the current editor entity from the route.
     $editor = NULL;
-    if ($route_match = \Drupal::routeMatch()) {
+    $route_match = \Drupal::routeMatch();
+    if ($route_match instanceof RouteMatchInterface) {
       $editor = $route_match->getParameter('editor');
     }
 

--- a/ckeditor_ai_agent.module
+++ b/ckeditor_ai_agent.module
@@ -9,6 +9,8 @@ use Drupal\ckeditor5\Plugin\CKEditor5PluginDefinition;
 
 /**
  * Implements hook_ckeditor5_plugin_info_alter().
+ *
+ * @phpstan-param \Drupal\ckeditor5\Plugin\CKEditor5PluginDefinition[] $plugin_definitions
  */
 function ckeditor_ai_agent_ckeditor5_plugin_info_alter(array &$plugin_definitions): void {
   if (isset($plugin_definitions['ckeditor_ai_agent_ai_agent'])) {

--- a/src/AiAgentConfigurationManager.php
+++ b/src/AiAgentConfigurationManager.php
@@ -33,7 +33,7 @@ class AiAgentConfigurationManager {
    * @param \Drupal\editor\Entity\Editor|null $editor
    *   The editor entity.
    *
-   * @return array
+   * @return array<string, array<string, mixed>>
    *   The CKEditor configuration.
    */
   public function getCkEditorConfig(?Editor $editor = NULL): array {

--- a/src/Form/AiAgentFormTrait.php
+++ b/src/Form/AiAgentFormTrait.php
@@ -71,7 +71,9 @@ trait AiAgentFormTrait {
       '#type' => 'select',
       '#title' => $this->t('AI Model'),
       '#options' => $getSelectOptions($model_options),
-      '#description' => $this->t('Select AI model' . ($is_plugin ? ' or use global settings.' : '.')),
+      '#description' => $this->t('@description', [
+        '@description' => 'Select AI model' . ($is_plugin ? ' or use global settings.' : '.'),
+      ]),
       '#default_value' => $getConfigValue('model'),
     ];
 

--- a/src/Form/AiAgentFormTrait.php
+++ b/src/Form/AiAgentFormTrait.php
@@ -329,7 +329,7 @@ trait AiAgentFormTrait {
           '#title' => $this->t('@label Override', ['@label' => $label]),
           '#default_value' => $getConfigValue("prompt_settings.overrides.$key"),
           '#placeholder' => $default_rules[$key] ?? '',
-          '#description' => $this->t('Override the default @label rules. Leave empty to use the default values shown above.', ['@label' => strtolower($label)]),
+          '#description' => $this->t('Override the default @label rules. Leave empty to use the default values shown above.', ['@label' => strtolower((string) $label)]),
           '#rows' => 6,
         ];
 
@@ -337,7 +337,7 @@ trait AiAgentFormTrait {
           '#type' => 'textarea',
           '#title' => $this->t('@label Additions', ['@label' => $label]),
           '#default_value' => $getConfigValue("prompt_settings.additions.$key"),
-          '#description' => $this->t('Add custom @label rules that will be appended to the defaults.', ['@label' => strtolower($label)]),
+          '#description' => $this->t('Add custom @label rules that will be appended to the defaults.', ['@label' => strtolower((string) $label)]),
           '#rows' => 4,
         ];
       }

--- a/src/Form/AiAgentFormTrait.php
+++ b/src/Form/AiAgentFormTrait.php
@@ -116,8 +116,8 @@ trait AiAgentFormTrait {
       $elements['advanced_settings']['tokens'][$field] = [
         '#type' => 'number',
         '#title' => $this->t('@title', ['@title' => $formatMachineNameAsTitle($field)]),
-        '#description' => $this->t('Maximum number of tokens for @type. If not set, uses model\'s maximum limit',
-                ['@type' => str_contains($field, 'output') ? 'AI response' : 'combined prompt and context']),
+        '#description' => $this->t("Maximum number of tokens for @type. If not set, uses model's maximum limit",
+          ['@type' => str_contains($field, 'output') ? 'AI response' : 'combined prompt and context']),
         '#min' => 1,
         '#default_value' => $getConfigValue("tokens.$field"),
       ];

--- a/src/Form/AiAgentFormTrait.php
+++ b/src/Form/AiAgentFormTrait.php
@@ -15,10 +15,10 @@ trait AiAgentFormTrait {
    * @param mixed $config
    *   Configuration object or array.
    *
-   * @return array
+   * @return array<string, mixed>
    *   The form elements.
    */
-  protected function getCommonFormElements($is_plugin = FALSE, $config = NULL) {
+  protected function getCommonFormElements($is_plugin = FALSE, $config = NULL): array {
     $elements = [];
 
     // Initialize config based on context.
@@ -285,15 +285,20 @@ trait AiAgentFormTrait {
     ];
 
     // Add prompt settings.
-    $this->addPromptSettings($elements, $is_plugin, $config, $getConfigValue);
+    $this->addPromptSettings($elements, $getConfigValue);
 
     return $elements;
   }
 
   /**
    * Adds prompt settings to the form elements.
+   *
+   * @param array<string, mixed> $elements
+   *   List of common form elements.
+   * @param \Closure $getConfigValue
+   *   Helper function to get config value based on context.
    */
-  protected function addPromptSettings(&$elements, $is_plugin, $config, $getConfigValue) {
+  protected function addPromptSettings(array &$elements, \Closure $getConfigValue): void {
     $elements['prompt_settings'] = [
       '#type' => 'details',
       '#title' => $this->t('Prompt Settings'),

--- a/src/Form/AiAgentFormTrait.php
+++ b/src/Form/AiAgentFormTrait.php
@@ -40,6 +40,9 @@ trait AiAgentFormTrait {
             : $options;
     };
 
+    // Helper function for formatting field names as titles.
+    $formatMachineNameAsTitle = fn($title) => str_replace('_', ' ', ucfirst($title));
+
     // Basic Settings.
     $elements['basic_settings'] = [
       '#type' => 'details',
@@ -112,7 +115,7 @@ trait AiAgentFormTrait {
     foreach ($token_fields as $field) {
       $elements['advanced_settings']['tokens'][$field] = [
         '#type' => 'number',
-        '#title' => $this->t(str_replace('_', ' ', ucfirst($field))),
+        '#title' => $this->t('@title', ['@title' => $formatMachineNameAsTitle($field)]),
         '#description' => $this->t('Maximum number of tokens for @type. If not set, uses model\'s maximum limit',
                 ['@type' => str_contains($field, 'output') ? 'AI response' : 'combined prompt and context']),
         '#min' => 1,
@@ -200,7 +203,7 @@ trait AiAgentFormTrait {
     foreach ($behavior_fields as $field) {
       $elements['behavior_settings'][$field] = [
         '#type' => 'select',
-        '#title' => $this->t(str_replace('_', ' ', ucfirst($field))),
+        '#title' => $this->t('@title', ['@title' => $formatMachineNameAsTitle($field)]),
         '#options' => $getSelectOptions($boolean_options),
         '#description' => $this->t('@desc', [
           '@desc' => 'Enable detailed logging for troubleshooting purposes.',

--- a/src/Form/AiAgentSettingsForm.php
+++ b/src/Form/AiAgentSettingsForm.php
@@ -35,7 +35,7 @@ class AiAgentSettingsForm extends ConfigFormBase {
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
+  public static function create(ContainerInterface $container): self {
     return new static(
       $container->get('extension.path.resolver')
     );
@@ -50,13 +50,18 @@ class AiAgentSettingsForm extends ConfigFormBase {
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-return string[]
    */
-  protected function getEditableConfigNames() {
+  protected function getEditableConfigNames(): array {
     return ['ckeditor_ai_agent.settings'];
   }
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
+   * @phpstan-return mixed[]
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
     $config = $this->config('ckeditor_ai_agent.settings');
@@ -91,8 +96,10 @@ class AiAgentSettingsForm extends ConfigFormBase {
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
    */
-  public function validateForm(array &$form, FormStateInterface $form_state) {
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
     parent::validateForm($form, $form_state);
 
     // Validate temperature range.
@@ -104,8 +111,10 @@ class AiAgentSettingsForm extends ConfigFormBase {
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
    */
-  public function submitForm(array &$form, FormStateInterface $form_state) {
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
     $config = $this->config('ckeditor_ai_agent.settings');
     $values = $form_state->getValues();
 

--- a/src/Form/ConfigMappingTrait.php
+++ b/src/Form/ConfigMappingTrait.php
@@ -12,6 +12,9 @@ trait ConfigMappingTrait {
    *
    * @param bool $is_plugin
    *   Whether this is for the plugin configuration.
+   *
+   * @return array<string, mixed>
+   *   The configuration mapping array.
    */
   protected function getConfigMapping(bool $is_plugin = FALSE): array {
     $base_mapping = [
@@ -95,6 +98,9 @@ trait ConfigMappingTrait {
 
   /**
    * Gets the prompt components list.
+   *
+   * @return string[]
+   *   The prompt components.
    */
   protected function getPromptComponents(): array {
     return [

--- a/src/Form/ConfigSetterTrait.php
+++ b/src/Form/ConfigSetterTrait.php
@@ -10,12 +10,12 @@ trait ConfigSetterTrait {
   /**
    * Sets configuration values with proper type casting.
    *
-   * @param array $values
+   * @param array<string, mixed> $values
    *   Form values to process.
-   * @param array $mapping
+   * @param array<string, mixed> $mapping
    *   Mapping of form keys to config keys with type information.
    *
-   * @return array
+   * @return array<string, mixed>
    *   Processed configuration array.
    */
   protected function processConfigValues(array $values, array $mapping): array {
@@ -35,8 +35,13 @@ trait ConfigSetterTrait {
 
   /**
    * Gets a nested array value using dot notation.
+   *
+   * @param array<string, mixed> $array
+   *   Array of form values.
+   * @param string $key
+   *   Form key used to extract the nested value.
    */
-  protected function getNestedValue(array $array, string $key) {
+  protected function getNestedValue(array $array, string $key): mixed {
     $keys = explode('.', $key);
     $value = $array;
 
@@ -53,7 +58,7 @@ trait ConfigSetterTrait {
   /**
    * Casts a value to the specified type.
    */
-  protected function castValue($value, string $type) {
+  protected function castValue(mixed $value, string $type): mixed {
     if (empty($value) && $value !== '0' && $value !== 0) {
       return NULL;
     }
@@ -78,6 +83,12 @@ trait ConfigSetterTrait {
 
   /**
    * Processes moderation settings.
+   *
+   * @param array<string, mixed> $values
+   *   Array of form values.
+   *
+   * @return array<string, mixed>
+   *   An array of moderation configuration properties.
    */
   protected function processModerationSettings(array $values): array {
     // Get moderation settings with defaults.
@@ -94,6 +105,12 @@ trait ConfigSetterTrait {
 
   /**
    * Processes prompt settings.
+   *
+   * @param array<string, mixed> $values
+   *   Array of form values.
+   *
+   * @return array<string, mixed>
+   *   An array of prompt configuration properties.
    */
   protected function processPromptSettings(array $values): array {
     $settings = [

--- a/src/Plugin/CKEditor5Plugin/AiAgent.php
+++ b/src/Plugin/CKEditor5Plugin/AiAgent.php
@@ -27,6 +27,8 @@ class AiAgent extends CKEditor5PluginDefault implements CKEditor5PluginConfigura
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-return array<string, mixed>
    */
   public function defaultConfiguration(): array {
     return [
@@ -74,6 +76,9 @@ class AiAgent extends CKEditor5PluginDefault implements CKEditor5PluginConfigura
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
+   * @phpstan-return array<string, mixed>
    */
   public function buildConfigurationForm(array $form, FormStateInterface $form_state): array {
     $config = $this->getConfiguration();
@@ -125,8 +130,10 @@ class AiAgent extends CKEditor5PluginDefault implements CKEditor5PluginConfigura
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
    */
-  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state): void {
     $values = $form_state->getValues();
 
     $this->configuration['aiAgent'] = $this->processConfigValues(
@@ -141,6 +148,9 @@ class AiAgent extends CKEditor5PluginDefault implements CKEditor5PluginConfigura
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $static_plugin_config
+   * @phpstan-return array<string, mixed>
    */
   public function getDynamicPluginConfig(array $static_plugin_config, EditorInterface $editor): array {
     $config = \Drupal::config('ckeditor_ai_agent.settings');
@@ -212,8 +222,10 @@ class AiAgent extends CKEditor5PluginDefault implements CKEditor5PluginConfigura
 
   /**
    * {@inheritdoc}
+   *
+   * @phpstan-param mixed[] $form
    */
-  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state): void {
     // Required by interface, but no validation needed.
   }
 

